x/gamm/pool-models/internal/cfmm_common/lp_test.go

```diff
@@ -0,0 +1,117 @@
+package cfmm_common_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/balancer"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/internal/cfmm_common"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/stableswap"
+	gammtypes "github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func TestCalcExitPool(t *testing.T) {
+	var (
+		two_asset_pool = &stableswap.Pool{
```

Comment:
Suggest using `NewStableswapPool(...)`

Reply:
The pools used in this test is not fully initialized because they only need to provide `GetTotalShares()` and `GetTotalPoolLiquidity()`. Fields like `Address` or `Id` are not relevant in this test so I don't want to use `NewStableswapPool()` which requires those fields.

Reply:
The reason why I suggest using `NewStableswapPool(...)` is because we are likely to keep adding initialization and validation logic in the constructor. Over time, it might start being invalid initializing a pool with or without certain parameters.

Using the constructor from the start would help to ensure that this initialization remains valid going forward. 

We can create:
```
const (
     testAddress = "testAddress"
     testId = "testId"
)
```
and provide that to `NewStableswapPool(...)`

Reply:
ohh that makes sense, I'll apply the change accordingly